.github/workflows/linux.yml

```diff
@@ -16,474 +16,488 @@ concurrency:
           path: ${{ env.OV_INSTALL_DIR }}
-
-      - name: Extract OpenVINO packages
-        run: |
-          pushd ${OV_INSTALL_DIR}
-            tar -xzf openvino_package.tar.gz -C ${OV_INSTALL_DIR} --strip-components=1
-          popd
-
-      - name: Set apt
+          merge-multiple: true
+          
+      - name: Download Archive Distribution Package
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: genai_archive_${{ matrix.build-type }}
+          merge-multiple: true
+          
+      - name: Extract Artifacts
+        run: pigz -dc genai.tar.gz | tar -xf - -C ${{ env.OV_INSTALL_DIR }}
+                    
+      - name: Build Samples - ${{ matrix.build-type }}
         run: |
-          echo 'Acquire::Retries "10";' | sudo tee -a /etc/apt/apt.conf.d/80-retries > /dev/null
-          echo 'APT::Get::Assume-Yes "true";' | sudo tee -a /etc/apt/apt.conf.d/81-assume-yes > /dev/null
-          echo 'APT::Get::Fix-Broken "true";' | sudo tee -a /etc/apt/apt.conf.d/82-fix-broken > /dev/null
-          echo 'APT::Get::no-install-recommends "true";' | sudo tee -a /etc/apt/apt.conf.d/83-no-recommends > /dev/null
+          source ${{ env.OV_INSTALL_DIR }}/setupvars.sh
+          cmake -DCMAKE_BUILD_TYPE=${{ matrix.build-type }} -S ${{ env.OV_INSTALL_DIR }}/samples/cpp/ -B ${{ env.BUILD_DIR }}
+          cmake --build ${{ env.BUILD_DIR }} --config ${{ matrix.build-type }} --parallel $(nproc)
+          cmake --install ${{ env.BUILD_DIR }} --config ${{ matrix.build-type }} --component samples_bin --prefix ${{ env.INSTALL_DIR }}
+        
+      - name: Pack Artifacts
+        run: tar -cvf - * | pigz > ${{ env.INSTALL_DIR }}/genai_samples.tar.gz
+        working-directory: ${{ env.INSTALL_DIR }}
 
-      - name: Install build dependencies
-        run: |
-          sudo ${OV_INSTALL_DIR}/install_dependencies/install_openvino_dependencies.sh
-          sudo apt-get install ccache
+      - name: Upload Samples Build Package
+        if: ${{ always() }}
+        uses: actions/upload-artifact@b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882 # v4.4.3
+        with:
+          name: genai_samples_${{ matrix.build-type }}
+          path: ${{ env.INSTALL_DIR }}/*.tar.gz
+          if-no-files-found: 'error'
+          
+  genai_tests_archive:
+    name: Python Tests (archive) - ${{ matrix.build-type }} - ${{ matrix.test.name }}
+    needs: [ openvino_download, genai_build_cmake ]
+    strategy:
+      fail-fast: false
+      matrix:
+        build-type: [Release]
+        test:
+          - name: 'test_chat_generate_api.py::test_set_chat_template'
+            cmd: 'tests/python_tests/test_chat_generate_api.py::test_set_chat_template'
+          - name: 'Whisper Generate API'
+            cmd: 'tests/python_tests/test_whisper_generate_api.py -k test_smoke'
+    timeout-minutes: 45
+    defaults:
+      run:
+        shell: bash
+    runs-on: aks-linux-2-cores-8gb-staging
```

Comment:
I think that the `staging` part could be dropped since we moved to the new cluster.

Reply:
fixed